src/scielo/bin/xml/app_modules/generics/dbm/dbm_isis.py

```diff
@@ -52,148 +40,139 @@ def _format_file(self, records):
+        """
+        Formata o registro
+        """
+        if record:
+            items = []
+            for tag_i in sorted([int(s) for s in record.keys()]):
                 tag = str(tag_i)
-                data = record.get(tag)
-                if data is not None and len(data) > 0:
-                    r.append(self.tag_data(tag, data))
-                else:
-                    res = self.tag_data(tag, data)
-                    if res != u'' and res is not None:
-                        encoding.debugging('_format_record()', res)
-            result = u''.join(r)
-
-            if self.content_formatter is not None:
+                items.extend(self._tag_data(tag, record[tag]))
+            result = "".join(items)
+            if self.content_formatter:
                 result = self.content_formatter(result)
-        return result
-
-    def tag_data(self, tag, data):
+            return result
+        return ""
+
+    def _tag_data(self, tag, data):
+        """
+        Cria os campos com respectivos conteúdos
+        """
+        if not data:
+            return []
         occs = []
         if not isinstance(data, list):
             data = [data]
         for item in data:
-            occs.append(self.tag_occ(tag, item))
-
-        return u''.join(occs)
-
-    def tag_occ(self, tag, data):
-        s = u''
-        if isinstance(data, tuple):
-            encoding.debugging('tag_occ()', tag)
-            encoding.debugging('tag_occ()', data)
-        elif isinstance(data, dict):
-            s = self.tag_content(tag, self.format_subfields(data))
-        else:
-            s = self.tag_content(tag, format_value(data))
-        return s
-
-    def format_subfield(self, subf, subf_value):
-        res = u''
-        if subf in 'abcdefghijklmnopqrstuvwxyz123456789':
-            res = '^' + subf + change_circ(format_value(subf_value))
-        elif subf != '_':
-            encoding.debugging('format_subfield()', ('ERR0', subf, subf_value))
-        return res
-
-    def format_subfields(self, subf_and_value_list):
-        first = u''
-        value = u''
-        values = []
+            occs.append(self._tag_occ(tag, item))
+        return occs
+
+    def _tag_occ(self, tag, data):
+        """
+        Cria cada ocorrência de um dado campo
+        """
+        if not data:
+            return ""
         try:
-            first = subf_and_value_list.get('_', u'') or u''
-            values = sorted([self.format_subfield(k, v) for k, v in subf_and_value_list.items() if v is not None and len(v) > 0])
-            value = u''.join(values)
-        except Exception as e:
-            encoding.report_exception('format_subfields()', e, subf_and_value_list)
-            encoding.report_exception('format_subfields()', e, (first, values))
+            data = {"_": data + ""}
+        except TypeError:
+            if not isinstance(data, dict):
+                raise TypeError("IDFile.tag_occ expects dict or str")
```

Comment:
@robertatakenaka, uma última dúvida: Se a variável `data` for um dicionário você tentará converte-la em uma lista de dicionários, é isso?

Reply:
data é o conteúdo de um campo. 
Se é str, significa que o valor da str é o valor do campo.
Se for dict, então é um campo com subcampos.

Reply:
Mas quando o dado é do tipo _dict_, ele é transformado em outra coisa?

Reply:
Na linha 92.